src/_pytest/fixtures.py

```diff
@@ -1060,14 +1066,14 @@ def _parse_fixture_args(callable_or_scope, *args, **kwargs):
 
 
 def fixture(
-    callable_or_scope=None,
+    callable_or_scope: Union[Callable[..., Any], str] = None,
     *args,
-    scope="function",
-    params=None,
-    autouse=False,
-    ids=None,
-    name=None
-):
+    scope: str = "function",
+    params: Optional[Iterable[Any]] = None,
+    autouse: bool = False,
+    ids: Optional[List[str]] = None,
+    name: Optional[str] = None
+) -> Callable[..., Any]:
```

Comment:
shouldn't this return `Callable[[T], T]` as it gives back a trivial decorator?

Reply:
I'm still learning how to annotate things. What is `T` in this case?

Reply:
you'll need to add `T = TypeVar('T')` -- it's a generic parameter

Reply:
OK, can do. I'm a bit sketchy on the difference between this vs `Any`.